src/firefly/js/templates/router/RoutedApp.jsx

```diff
@@ -0,0 +1,105 @@
+import React, {useEffect} from 'react';
+import {cloneDeep} from 'lodash';
+import {useOutlet} from 'react-router-dom';
+
+import {FireflyLayout} from 'firefly/templates/common/FireflyLayout.jsx';
+import {IrsaFooterSmall} from 'firefly/ui/IrsaFooter.jsx';
+import {useDropdownRoute} from './RouteHelper.jsx';
+import {DropDownContainer} from 'firefly/ui/DropDownContainer.jsx';
+import {startTTFeatureWatchers} from 'firefly/templates/common/ttFeatureWatchers.js';
+import {dispatchNotifyRemoteAppReady, dispatchOnAppReady, dispatchSetMenu} from 'firefly/core/AppDataCntlr.js';
+import {HydraLanding} from 'firefly/templates/hydra/HydraViewer.jsx';
+import {setIf as setIfUndefined} from 'firefly/util/WebUtil.js';
+import {Slot} from 'firefly/ui/SimpleComponent.jsx';
+
+
+/*
+
+An application that uses FireflyLayout and its common components such as DropDownContainer, Menu, Banner, LandingPage, etc.
+For a better understanding of its responsibilities, refer to the FireflyLayout documentation.
+
+A significant distinction of RoutedApp lies in its rendering mechanism. Typically, routed apps render based on their paths.
+This functionality remains intact, extending up to what FireflyLayout is responsible for.
+However, RoutedApp also support Firefly's layout controller (LayoutCntlr), which controls the display of Menu state, dropdown state,
+and the MainPanel (results) of the application.
+RoutedApp achieves this by synchronizing and mapping Firefly's layout information with the application's defined routes.
+
+
+Several key concepts to understand when using RoutedApp:
+
+1. The route define what get rendered.  Therefore, RoutedApp should be at the root of all routes.
+2. React Router introduces the notion of child routes and the element rendered within the parent's element, referred as 'outlet' in the code.
+   Examine this code snippet to understand the differences between an outlet and children.
+
+    const routes = [
+        {
+            path: '/',
+            element: <RoutedApp />,
+            children: [
+                {
+                    path: '/searchByPosition',
+                    element: <SearchByPosition/>
+                },
+            ]
+        },
+        {
+            path: '/somethingElse',
+            element: <RoutedApp> <SomethingElse/> </RoutedApp>,
+        }
+    ];
+
+In this example, `<SearchByPosition/>` serves as an outlet of RoutedApp since its route is defined as a child of RoutedApp.
+'<SomethingElse/>' appears as children because it's declared inside of RoutedApp like regular react component.
```

Comment:
Thanks for the documentation, very helpful in understanding how route based apps work. Also, these changes now have a major improvement for Euclid (fixes an old bug): If you refreshed the page while on an endpoint like `/euclid/search-region-pos `, earlier this would just show an empty page, this is fixed now for all routes!